### PR TITLE
chore(ci): add codecov

### PR DESCRIPTION
**Pull request by rahul2393**
### 1. Explain what the PR does

Fixes: https://github.com/aquasecurity/tracee/issues/152

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -224,6 +224,11 @@ jobs:
       - name: Run Unit Tests
         run: |
           make test-unit
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.0.1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (geyslan) [inline .github/workflows/pr.yaml:230]: Is this secret already set?

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -224,6 +224,11 @@ jobs:
       - name: Run Unit Tests
         run: |
           make test-unit
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.0.1
```
Comment (geyslan) [inline .github/workflows/pr.yaml:228]: Would you mind using this pattern?

`...@hash # vX.X.X`

https://github.com/aquasecurity/tracee/blob/00598de5cd314b6871e06d6c561f6a7708923f04/.github/workflows/pr.yaml#L128

**Review by geyslan — COMMENTED** [review]
@rahul2393 I've put some thoughts. 👍🏼

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (rahul2393) [reply .github/workflows/pr.yaml:230]: @yanivagman to confirm.

Comment (rahul2393) [reply .github/workflows/pr.yaml:230]: Changes are already tested on fork repo https://github.com/rahul2393/tracee/pull/6

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -224,6 +224,11 @@ jobs:
       - name: Run Unit Tests
         run: |
           make test-unit
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@e0b68c6749509c5f83f984dd99a76a1c1a231044 #v4.0.1
```
Comment (geyslan) [inline .github/workflows/pr.yaml:228]: Nice!

The latest version is [v4.4.1](https://github.com/codecov/codecov-action/releases/tag/v4.4.1). Is there a reason for not using it? If so, please insert a comment explaining it.

**Pull request closed without merging**